### PR TITLE
[CDEC-658][CDEC-659] Updated CHANGELOG.md with [CDEC-658] and [CDEC-659]

### PR DESCRIPTION
**Pull request by coot**
## Description
Update CHANGELOG.md file.

## Linked issues

https://iohk.myjetbrains.com/youtrack/issue/CDEC-658
https://iohk.myjetbrains.com/youtrack/issue/CDEC-659

## Type of change

- [ ] 🐞 Bug fix (non-breaking change which fixes an issue)
- [ ] 🛠 New feature (non-breaking change which adds functionality)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)
- [ ] 🏭 Refactoring that does not change existing functionality but does improve things like code readability, structure etc
- [ ] 🔨 New or improved tests for existing code
- [ ] ⛑ git-flow chore (backport, hotfix, etc)

## Developer checklist

- [ ] I have read the [style guide](https://github.com/input-output-hk/cardano-sl/blob/develop/docs/style-guide.md) document, and my code follows the code style of this project.
- [ ] If my code deals with exceptions, it follows the [guidelines](https://github.com/input-output-hk/cardano-sl/blob/develop/docs/exceptions.md).
- [ ] I have updated any documentation accordingly, if needed. Documentation changes can be reflected in opening a PR on [cardanodocs.com](https://github.com/input-output-hk/cardanodocs.com), amending the inline [Haddock](https://www.haskell.org/haddock/) comments, any relevant README file or one of the document listed in the [docs](https://github.com/input-output-hk/cardano-sl/tree/develop/docs) directory.
- [ ] CHANGELOG entry has been added and is linked to the correct PR on GitHub.

## Testing checklist

- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

## How to merge

Send the message `bors r+` to merge this PR. For more information, see
[`docs/how-to/bors.md`](https://github.com/input-output-hk/cardano-sl/blob/develop/docs/how-to/bors.md).

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -5,7 +5,8 @@
 ### Fixes
 
 - Relocate fee sanity check and make it relative to each transaction (rather than absolute) ([CO-446](https://iohk.myjetbrains.com/youtrack/issue/CO-446) [#3993](https://github.com/input-output-hk/cardano-sl/pull/3993)
-
+- Change how block retrieval worker's conversations are organised, so that the worker can recover from IOExceptions ([CDEC-658](https://iohk.myjetbrains.com/youtrack/issue/CDEC-658) [#3994](https://github.com/input-output-hk/cardano-sl/pull/3994))
+- Limit the rate of dns queries rate to avoid reaching the limit of open files, try resolve dns in ntp client every 30s rather than once ([CDEC-659](https://iohk.myjetbrains.com/youtrack/issue/CDEC-658) [#4008](https://github.com/input-output-hk/cardano-sl/pull/4008))
```
Comment (avieth) [inline CHANGELOG.md]: Limit the rate of dns queries ~rate~ to avoid reaching the limit of open files; try to resolve dns in ntp client every 30s rather than *only* once.

File: CHANGELOG.md [hunk 2]
```diff
@@ -5,7 +5,8 @@
 ### Fixes
 
 - Relocate fee sanity check and make it relative to each transaction (rather than absolute) ([CO-446](https://iohk.myjetbrains.com/youtrack/issue/CO-446) [#3993](https://github.com/input-output-hk/cardano-sl/pull/3993)
-
+- Change how block retrieval worker's conversations are organised, so that the worker can recover from IOExceptions ([CDEC-658](https://iohk.myjetbrains.com/youtrack/issue/CDEC-658) [#3994](https://github.com/input-output-hk/cardano-sl/pull/3994))
```
Comment (avieth) [inline CHANGELOG.md]: I'd reword this as

"Correction to block retrieval concurrency and exception handling, so that it will recover in case of certain network failure conditions."

File: CHANGELOG.md (hunk 1 above)
Comment (coot) [reply CHANGELOG.md]: I updated the PR.
